aspnetcore/blazor/performance/webassembly-event-pipe-diagnostics.md

```diff
@@ -0,0 +1,109 @@
+---
+title: ASP.NET Core Blazor WebAssembly Event Pipe diagnostics
+author: guardrex
+description: Learn about Event Pipe diagnostics and how to get a Garbage Collector heap dump in ASP.NET Core Blazor WebAssembly apps.
+monikerRange: '>= aspnetcore-10.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 05/02/2025
+uid: blazor/performance/webassembly-event-pipe
+---
+# ASP.NET Core Blazor WebAssembly Event Pipe diagnostics
+
+<!-- UPDATE 10.0 - Activate ...
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+-->
+
+This article describes Event Pipe diagnostic tools, counters, and how to get a Garbage Collector heap dump in Blazor WebAssembly apps.
+
+## Prerequisite
+
+Install the [.NET WebAssembly build tools](xref:blazor/tooling/webassembly#net-webassembly-build-tools):
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+## EventPipe profiler
+
+[EventPipe](/dotnet/core/diagnostics/eventpipe) is a runtime component used to collect tracing data, similar to [ETW](/windows/win32/etw/event-tracing-portal) and [perf_events](https://wikipedia.org/wiki/Perf_%28Linux%29).
+
+Collect CPU counters for 60 seconds with `collectCpuSamples(durationSeconds)`:
+
+```javascript
+globalThis.getDotnetRuntime(0).collectCpuSamples({durationSeconds: 60});
+```
+
+## GC (Garbage Collector) dumps
+
+Manual testing:
+
+* Browser developer tools: Download the `.json` output file, open the file in Visual Studio, and find the expected classes.
+* [`dotnet-gcdump` (`collect`/convert` options)](/dotnet/core/diagnostics/dotnet-gcdump): To view the captured GC dump files, see [View the GC dump captured from dotnet-gcdump](/dotnet/core/diagnostics/dotnet-gcdump#view-the-gc-dump-captured-from-dotnet-gcdump).
+
+Collect a GC (Garbage Collector) dump of the live .NET process with `collectGcDump`:
+
+```javascript
+globalThis.getDotnetRuntime(0).collectGcDump();
+```
+
+## Counters trace
+
+Manual testing:
+
+* Browser developer tools: Download the `.json` output file, open the file in Visual Studio, and find the expected counters.
+* [`dotnet-counters collect`](/dotnet/core/diagnostics/dotnet-counters): Open the `.csv`/`.json` output file in Visual Studio and find the expected counters.
+
+Collect diagnostic counters for 60 seconds with `collectPerfCounters(durationSeconds)`:
+
+```javascript
+globalThis.getDotnetRuntime(0).collectPerfCounters({durationSeconds: 60});
+```
+
+## .NET Core Diagnostics Client Library example
```

Comment:
Maybe move the whole paragraph with the msbuild properties up in the page, just after prerequisites, because you can;t really use the `globalThis.getDotnetRuntime(0).collectPerfCounters` before enabling that.
```